### PR TITLE
New doc describing OIDC integration

### PR DESCRIPTION
**Pull request by thomasvn**
Full credit to @biancaburtoiu. This is a migration of her Google Doc to a Markdown doc.

This doc complements the existing [SSO/SAML doc](https://github.com/kubecost/docs/blob/main/user-management.md). It is titled similarly to that doc and will be placed in the same section.

In this Markdown doc, I left out detailed instructions for how to integrate with the Keycloak Identity Provider, since that currently references another internal Google Doc.

### CODE REVIEW

File: user-management-oidc.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+Please refer to [Azure documentation](https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc#send-the-sign-in-request).
+
+### Google OAuth 2.0
+
+Please refer to [Google OAuth 2.0 documentation](https://developers.google.com/identity/openid-connect/openid-connect#authenticatingtheuser).
+
+### Auth0
+
+Please refer to [Auth0 documentation](https://auth0.com/docs/get-started/authentication-and-authorization-flow/add-login-auth-code-flow).
```
Comment (bstuder99) [inline user-management-oidc.md:31]: A more concise way of formatting this (as we recommend consulting respective docs for each provider) would be to list them as bullet points with the docs embedded as a hyperlink in each name. A line could be added under "##Supported Identity Providers" explaining that each link goes to the providers' docs.

**Comment by bstuder99** [comment]
Made a couple changes, other than my hanging comment above, LGTM

**Comment by thomasvn** [comment]
Thanks for the voice/style edits @bstuder99! I've additionally made the identity provider section more concise

File: user-management-oidc.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+If the domain is configured alongside the access token, then requests should contain the JWT ID token, either:
+
+* as a cookie named `id_token`, or
+* as part of an `Identification` header.
+
+The JWT ID token must contain a field (claim) named `hd` with the desired domain value. We verify that the token has been properly signed (using provider certificates) and has not expired before processing the claim.
+
+Removing a previously set Helm value is done by setting `.Values.oidc.hostedDomain = ""`. Simply removing the hostedDomain field from the helm config will not work at the moment.
+
+Validate that the config has been correctly removed by checking that `hostedDomain=""` in the **/var/configs/oidc/oidc.json** file inside the cost-model container.
```
Comment (biancaburtoiu) [inline user-management-oidc.md:49]: This specific sentence referencing the container file check was a note for the Support team, I'm not entirely sure whether we want to leave that public. Let me know what you think!

File: user-management-oidc.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+For Google OAuth 2.0, the domain to match the hostedDomain helm parameter is the hd parameter in the id_token field of the token endpoint response.
```
Comment (biancaburtoiu) [inline user-management-oidc.md:51]: I don't think this sentence is necessary anymore. The same information already exists above.

File: user-management-oidc.md (hunk 2 above)
Comment (thomasvn) [reply user-management-oidc.md:49]: I think there's some value in leaving this info public to help users troubleshoot. However I'll reword it!

File: user-management-oidc.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+> **Note**: OIDC capabilities are only included with a Kubecost Enterprise Subscription.
+
+## Helm configuration
+
+The OIDC integration in Kubecost is fulfilled via the helm configuration parameters `.Values.oidc` [as shown here](https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/values.yaml).
+
+## Supported identity providers
+
+Please refer to the following references to find out more about how to configure the Helm parameters to suit each OIDC identiy provider integration.
+
```
Comment (biancaburtoiu) [inline user-management-oidc.md:13]: There is an information gap here as we have no public Keycloak documentation to reference. I had a brief look over their pages and I couldn't find a valid resource to share. On the other hand, we have this [internal Keycloak doc](https://docs.google.com/document/d/10XtJhWlYD0TcjN-Tohk38eXWmUwJHJsomOti7uqiAJ8/edit#).

We might want to revisit this section altogether and deciding whether we want to share these external resources in our public docs in the first place, or to leave them internal to our Support team. 

Do you have any thoughts here? @thomasvn 
cc @kaelanspatel

Comment (kaelanspatel) [reply user-management-oidc.md:13]: This is more of a product decision, tbh. While I'm in favor of making config as easy as possible, where did we land officially on making exact setup public? 

Maybe @AjayTripathy can say?

Comment (AjayTripathy) [reply user-management-oidc.md:13]: I think it's best to document thoroughly and make the setup public but note that this is enterprise-only. That sound good?

Comment (bstuder99) [reply user-management-oidc.md:13]: I would agree. We've documented Enterprise only features like that before. Need me to make an edit?

File: user-management-oidc.md [hunk 5]
```diff
@@ -11,14 +11,44 @@ The OIDC integration in Kubecost is fulfilled via the helm configuration paramet
...
+1. Create a new [Keycloak Realm](https://www.keycloak.org/getting-started/getting-started-kube#_create_a_realm).
+2. Navigate to "Realm Settings" -> "General" -> "Endpoints" -> "OpenID Endpoint Configuration" -> "Clients".
+3. Click "Create" to add Kubecost to the list of clients. Define a `clientID`. Ensure the "Client Protocol" is set to `openid-connect`.
+4. Click on your newly created client, then go to "Settings".
+   1. Set "Access Type" to `confidential`.
+   2. Set "Valid Redirect URIs" to `http://your-kubecost-address/model/oidc/authorize`.
+   3. Set "Base URL" to `http://your-kubecost-address`.
+
+The [`.Values.oidc`](https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/values.yaml) for Keycloak should be as follows:
```
Comment (srpomeroy) [inline user-management-oidc.md:33]: Suggest we use a permalink to GitHub files.
ex: https://github.com/kubecost/cost-analyzer-helm-chart/blob/93d3255870766d236628359a969b7691450d0095/cost-analyzer/values.yaml#L194-L202

Comment (srpomeroy) [reply user-management-oidc.md:33]: Would apply to the same link near the top of the page as well.

Comment (thomasvn) [reply user-management-oidc.md]: @srpomeroy Updated to become permalinks!

File: user-management-oidc.md (hunk 4 above)
Comment (thomasvn) [reply user-management-oidc.md:13]: @biancaburtoiu . I've added the Keycloak setup information. Could you help to validate its correctness?

Comment (biancaburtoiu) [reply user-management-oidc.md:13]: this lgtm, @kaelanspatel wdyt about these keycloak setup instructions refactored for the public docs?

Comment (kaelanspatel) [reply user-management-oidc.md:13]: @biancaburtoiu @thomasvn LGTM! Tiny thing, I think adding "dummy" values (e.g. `clientID: "<your client ID>"`  vs `clientID: ""`) makes the helm config section clearer at a glance. Seeing "the `Values.oidc` for Keycloak should be as follows" and having a whole bunch of blank values threw me off for a second.

Obviously a nit, great stuff in any case.

Comment (thomasvn) [reply user-management-oidc.md:13]: @biancaburtoiu @kaelanspatel. Thanks for the review! I've added in the dummy values, and also capitalized the dummy values to make it more pronounced that they need to be replaced.

**Comment by thomasvn** [comment]
Requesting final approval from @kubecost/documentation-team ! I've received technical validation from @biancaburtoiu and @kaelanspatel.

I've also rebased with the main branch (although now I'm realizing I should've just merged with main to make it less messy 😥)

**Review by srpomeroy — APPROVED** [review]

**Pull request merged**